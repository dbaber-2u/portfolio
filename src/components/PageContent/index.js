import React from 'react';
import AboutMe from '../AboutMe';
import Portfolio from '../Portfolio';

function PageContent({currentNavigation}) {
    switch(currentNavigation.name)
    {
      //case "Resume": return (<Resume />)
      //case "Contact": return (<Contact />)
      case "Portfolio": return (<Portfolio />)
      default: return (<AboutMe />)
    }
}

export default PageContent;