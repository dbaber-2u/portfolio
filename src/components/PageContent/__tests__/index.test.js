import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import PageContent from '..';


const navigations = [
    { "name": "About Me" },
    { "name": "Contact" },
    { "name": "Portfolio" },
    { "name": "Resume" }
];

afterEach(cleanup);



describe('PageContent component', () => {
    // baseline test
    //it('renders', () => {
    //    render(<PageContent navigation={[currentNavigation]} currentNavigation={currentNavigation} />);
    //});

    navigations.map((curNav, i) => (
        it('renders ${curNav.name}', () => {
            render(<PageContent navigation={navigations} currentNavigation={curNav} />);
        })
    ));

    // snapshot test
    it('matches snapshot', () => {
        const { asFragment } = render(<PageContent currentNavigation={navigations[0]} />);

        expect(asFragment()).toMatchSnapshot();
    });
})