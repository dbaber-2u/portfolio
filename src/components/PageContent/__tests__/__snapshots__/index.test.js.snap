// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContent component matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <div>
      <img
        alt="me profile pic"
        class="img-thumbnail mx-1"
        src="ProfilePic.jpg"
      />
    </div>
    <p>
      This is a little bit about me:
    </p>
    <p />
    <ul>
      <li>
        Software Engineer for 20+ years
      </li>
      <li>
        Lego Enthusiast
      </li>
      <li>
        Husband
      </li>
      <li>
        Father to 9
      </li>
    </ul>
  </section>
</DocumentFragment>
`;
